File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int sco : scoville){
            pq.add(sco);
        }
        int answer =0;
        while(pq.peek()<K){
            if(pq.size() < 2)
                return -1;
            int first = pq.poll();
            int second = pq.poll();
            pq.add(first+second*2);
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq as hq

def solution(scoville, K):
    answer = 0
    hq.heapify(scoville)
    
    while True:
        minVal = hq.heappop(scoville)
        if minVal >= K:
            break
        elif len(scoville) == 0:
            return -1
        secondVal = hq.heappop(scoville)
        hq.heappush(scoville, minVal + secondVal*2)
        answer+=1
    return answer 